src/DynamoCore/Graph/Workspaces/HomeWorkspaceModel.cs

```diff
@@ -331,9 +332,16 @@ internal override void RequestRun()
         {
             base.RequestRun();
 
-            if (RunSettings.RunEnabled && RunSettings.RunType != RunType.Manual)
+            if (RunSettings.RunType != RunType.Manual)
             {
-                Run();
+                // TODO for Dynamo 3.0: The boolean "executingTask" that is used here is a make-do fix.
+                // We will be needing a separate variable(boolean) to check for RunEnabled flag from external applications and
+                // not confuse it with the internal flag RunEnabled which is associated with the Run button in Dynamo. 
+                // Make this RunSettings.RunEnabled private, introduce the new flag and remove the "executingTask" variable. 
+                if (RunSettings.RunEnabled || executingTask)
```

Comment:
should it be `!executingTask`?

Reply:
No, because this boolean tracks if the previous run is completely executed or not. In the case, where OnUpdateGraphCompleted event is not completed, we want to trigger the next runs. In the end the executing task will be set to false.

Reply:
I see, since we are using `||`, normal case it will just be dominated by `RunSettings.RunEnabled` which is always true. But once `RunEnabled ` become false, `run()` can't be called at all